### DISCUSSION TITLE
🐞 Module names with single-dotted  characters cause "main object not found"

### DISCUSSION THREAD

**Discussion by SKalt** [dagger/dagger#12297] [Help]
### What is the issue?

I have a go module with a weird name with single dotted characters in it (e.g. "m.a.s.h"). I was running through the setup script when I encountered a weird error, "unable to find main object". I determined this error only occurs when there are single-dotted characters in the name field of `dagger.json`. I suspect the culprit is a mismatch between the arguments informing the [generated go struct name](https://github.com/dagger/dagger/blob/1bc2cffabc80e614afea9dc0155a0e77af1aa4c4/cmd/codegen/generator/go/generator.go#L373)

```go
moduleStructName := strcase.ToCamel(g.Config.ModuleName)
```

and the [gql object name](https://github.com/dagger/dagger/blob/1bc2cffabc80e614afea9dc0155a0e77af1aa4c4/core/gqlformat.go#L25)

```go
func gqlObjectName(name string) string {
	return strcase.ToCamel(name)
}
```

### Dagger version

engine 6cf72caa80eb (version v0.11.2)

### Steps to reproduce

Full reproduction script at [https://github.com/SKalt/weird.name.repro/blob/main/repro.sh](https://github.com/SKalt/weird.name.repro/blob/main/repro.sh). Relevant excerpt:

```sh
dagger init --name=w.eird.name.repro
dagger develop --sdk=go
dagger functions
```

### Log output

Full output logs at [https://github.com/SKalt/weird.name.repro/blob/main/output.txt](https://github.com/SKalt/weird.name.repro/blob/main/output.txt). Relevant excerpt:

```output
Error: main object not found
```

**Comment by helderco** [comment]
Yes, that error appears when there's no object in the module's code that match the module's name, taking into consideration naming convention conversions. Safest bet for module names is *kebab-case*.

However, the error can be improved here.

**Comment by rajatjindal** [comment]
for this issue, do folks recommend attempting to fix handling of this dotted separated name or adding a better error msg when this happens?

**Comment by helderco** [comment]
I think it's best to transform to a valid one automatically. So if the directory's name is `m.a.s.h`, transform to `m-a-s-h` (which translates to an object name of `MASH`), or to `mash`. 

I don't know if it's best to remove invalid characters or to replace with a dash.

If there's an explicit `--name=` flag, we could error, but if we already transform from the directory name I think we should also transform from the user input.

We should error if user edited `dagger.json` manually and it's an invalid name. So only transform in the CLI when saving to `dagger.json`.

**Comment by rajatjindal** [comment]
with the changes in https://github.com/dagger/dagger/pull/8122, this seems to be working mostly, except for Python, for which I had to fix to_pascal_case function in dagger python utils.

once that PR is merged, I will open the PR for this fix + the testcases to cover this scenario.

**Comment by fredleger** [comment]
seems that is not only with dots

I think i faced this one with the name of my simple python script: `hokk2mail`
I add to adjust the `dagger.json`  name entry to 'hook_2_mail' instead of the default generated `hook2mail` to make dagger call correctly the sample pipeline 'container-echo' with a newly dagger init project

running Python 3.13.0 in a virual env and dagger v0.15.3 on MacOS M1

![Image](https://github.com/user-attachments/assets/9f62cd29-49dd-49d0-b74c-66200a8a94ba)
